splunk-otel-android/src/main/java/com/splunk/rum/RumInitializer.java

```diff
@@ -241,24 +247,65 @@ private SdkTracerProvider buildTracerProvider(
 
     //visible for testing
     SpanExporter buildExporter(ConnectionUtil connectionUtil) {
-        String endpoint = config.getBeaconEndpoint() + "?auth=" + config.getRumAccessToken();
         if (!config.isDebugEnabled()) {
             //tell the Zipkin exporter to shut up already. We're on mobile, network stuff happens.
             // we'll do our best to hang on to the spans with the wrapping BufferingExporter.
             ZipkinSpanExporter.baseLogger.setLevel(Level.SEVERE);
             initializationEvents.add(new InitializationEvent("logger setup complete", timingClock.now()));
         }
-        SpanExporter zipkinSpanExporter = getCoreSpanExporter(endpoint);
+
+        if(config.isDiskBufferingDisabled()){
+            SpanExporter spanExporter = buildMemoryBufferingThrottledExporter(connectionUtil);
+            initializationEvents.add(new InitializationEvent("zipkin exporter initialized", timingClock.now()));
+            return spanExporter;
+        }
+        return buildStorageBufferingExporter(connectionUtil);
+    }
+
+    private SpanExporter buildStorageBufferingExporter(ConnectionUtil connectionUtil) {
+        Sender sender = OkHttpSender.newBuilder()
+                .endpoint(getEndpoint())
+                .build();
+        Path filesDir = Paths.get(application.getApplicationContext().getFilesDir().toURI());
+        Path spanFilesPath = filesDir.resolve("spans");
+        DiskToZipkinExporter diskToZipkinExporter = DiskToZipkinExporter.builder()
+                .connectionUtil(connectionUtil)
+                .sender(sender)
+                .spanFilesPath(spanFilesPath)
+                .build();
+        diskToZipkinExporter.startPolling();
+
+        SpanExporter diskBufferingExporter = getToDiskExporter();
+        SpanExporter spanExporter = config.decorateWithSpanFilter(diskBufferingExporter);
```

Comment:
This line is common to both disk & memory pipelines; how about moving it up to the `buildExporter` method?

Reply:
Yeah good call. There was something nagging me about this in my brain. The initialization event is also duplicated, so I've pulled that up too.